File: Geklaut/05_WTApp_Gutschier/src/main/java/htlvillach/gutschim/wtapp/models/Company.java
package htlvillach.gutschim.wtapp.models;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
public class Company {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String name;
  private String zipTown;
  private String street;
  private String phone;
  private String email;
  private String replyTo;
  private String comments;

  @OneToMany(mappedBy = "company", orphanRemoval = true, cascade = CascadeType.ALL)
  private List<Participation> participations;

}

File: KOTLIN/Wirtschaftstag/WTMobile/app/src/main/java/com/example/wtmobile/data/models/Company.kt
package com.example.wtmobile.data.models

import com.example.wtmobile.data.models.Participation

data class Company(
    var id: Int,
    var name: String,
    var zipTown: String,
    var street: String,
    var phone: String,
    var replyTo: String,
    var comments: String,
    var email: String,
) {
    override fun toString(): String {
        return "Company #$id\n$name in $zipTown, $street.\nContact: $replyTo\nEmail: $email\nPhone: $phone \n" +
                "Comments: $comments"
    }
}